metrics/metrics.go

```diff
@@ -39,32 +39,33 @@ var (
 		prometheus.CounterOpts{
 			Namespace: metricsNamespace,
 			Name:      "storage_indexer_update_index_success_total",
-			Help:      "A counter for updates of the cursor.",
+			Help:      "A counter for updates of the cursor in the storage indexer.",
 		},
 	)
 
 	StorageIndexerUpdateIndexErrorsTotal = prometheus.NewCounter(
 		prometheus.CounterOpts{
 			Namespace: metricsNamespace,
 			Name:      "storage_indexer_update_index_error_total",
-			Help:      "A counter for all the update index processes that finished with error.",
+			Help:      "A counter for all the update index processes that finished with error in the storage indexer.",
 		},
 	)
 
 	StorageIndexerUpdateIndexDurationSeconds = prometheus.NewHistogram(
 		prometheus.HistogramOpts{
 			Namespace: metricsNamespace,
 			Name:      "storage_indexer_update_index_duration_seconds",
-			Help:      "A histogram of latencies for update index processes run by the indexer.",
+			Help:      "A histogram of latencies for update index processes run by the storage indexer.",
 		},
 	)
 
-	StorageIndexerGetDurationSeconds = prometheus.NewHistogram(
+	IndexerGetDurationSeconds = prometheus.NewHistogramVec(
```

Comment:
Umm, I am wondering how useful is this metric. Looking at the sample metrics in the description, they seem to be all quite fast as these operations happen mostly in memory, and they end up all in the lower bucket.
We also have metrics in the middleware for `/search`, that may be more useful for final users, as it includes the time of the whole response.

So I wonder if maybe we can remove this metric. Or if we leave it, lower the boundaries of each bucket so they don't end up all in the lower bucket.

Reply:
Currently I don't have data about these get operations in the storage indexer. 
I was expecting that this metric would be more significant in that storage indexer than in the file system indexers. I was thinking that maybe those get ops against storage indexer would be slower (just hypothesis, I don't have data about this).

Reply:
Well, current implementations are only filtering a list in memory. But it is true that the operation in the storage indexer is in a mutex whose lock can have some wait time while the index is updated. We may revisit the bucket thresholds when we have more data.